app/uk/gov/hmrc/nrs/retrieval/controllers/NonrepRetrievalController.scala

```diff
@@ -19,61 +19,70 @@ package uk.gov.hmrc.nrs.retrieval.controllers
 import javax.inject.Singleton
 import com.google.inject.Inject
 import play.api.mvc._
-import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.http.{BadGatewayException, HeaderCarrier, HttpResponse, InternalServerException, NotFoundException}
 import uk.gov.hmrc.nrs.retrieval.connectors.NonrepRetrievalConnector
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 
 import scala.concurrent.ExecutionContext
 
 @Singleton()
-class NonrepRetrievalController @Inject()(val nonrepRetrievalConnector: NonrepRetrievalConnector,
-                                          override val controllerComponents: ControllerComponents)
-                                         (implicit ec: ExecutionContext)
-  extends BackendController(controllerComponents) {
+class NonrepRetrievalController @Inject()(
+  val nonrepRetrievalConnector: NonrepRetrievalConnector,
+  override val controllerComponents: ControllerComponents)(implicit ec: ExecutionContext)
+    extends BackendController(controllerComponents) {
 
-   def search() = Action.async { implicit request =>
+  val search: Action[AnyContent] = Action.async { implicit request =>
```

Comment:
can be a val

Reply:
Are you sure about it?
Won't it cause a bit unexpected behaviour of the service?

Reply:
No, the def or val returns an `Action` which is executed each time the route is accessed. The Action instance has a function which takes the request as a parameter. 

The def does this every time, the val does it once, but in this you get the same Action because there are no parameters to this method. The dynamic elements are on the request which is passed to the action on each execution. 

Happy to roll it back though if you're worried - its not necessary for this change.

Reply:
I get the difference between val and def.
If you are sure that you won't get an action persisted to the first request that go through the system then I'm fine with that